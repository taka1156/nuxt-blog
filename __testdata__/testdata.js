const dummyImgUrl = 'http://placehold.jp/150x150.png';
const MAX_NAV_DATA = 4;
const MAX_ARTICLE_DATA = 10;
const MAX_BADGE_DATA = 5;
const MAX_TOC_DATA = 5;

// factory
const dummyFactory = (num, fn) => {
  return [...new Array(num)].map((_, i) => fn(i + 1));
};

// template
const dummyNavTemplate = () => ({
  name: 'ダミーナビゲーション',
  to: '/test1',
  img: dummyImgUrl
});

const dummyBadgeTemplate = (i, name) => ({
  id: i,
  name: name,
  img: { url: dummyImgUrl }
});

const dummyClassficationTemplate = (i, name) => ({
  id: i,
  name: name,
  img: {
    url: dummyImgUrl
  }
});

const dummyTocTemplate = i => ({
  index: i,
  escapedText: 'この文章はダミーです。',
  anchor: `anchor_${i}`
});

const dummyMarkdown = `
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
`;

const dummyArticleTemplate = i => ({
  id: i,
  ...dummyDate,
  title:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
  summary:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
  body: dummyMarkdown,
  tags: dummyTagBadges.slice(0, i),
  category: dummyBadgeTemplate(1, 'ダミーカテゴリー')
});

// dummyData
const dummyImg = {
  imgUrl: dummyImgUrl,
  imgAlt: 'ダミーの画像'
};

const dummyDate = {
  createdAt: '2020-07-04T10:53:40.252Z',
  updatedAt: '2020-07-08T15:15:07.668Z'
};

const dummyRoutes = dummyFactory(MAX_NAV_DATA, () => dummyNavTemplate());

const dummyTagBadges = dummyFactory(MAX_BADGE_DATA, i =>
  dummyBadgeTemplate(i, 'ダミータグ')
);

const dummyClassificationCategory = dummyFactory(MAX_BADGE_DATA, i =>
  dummyClassficationTemplate(i, 'ダミーカテゴリー')
);

const dummyClassificationTag = dummyFactory(MAX_BADGE_DATA, i =>
  dummyClassficationTemplate(i, 'ダミータグ')
);

const dummyArticles = dummyFactory(MAX_ARTICLE_DATA, i => dummyArticleTemplate(i));

const dummyCategoryBadge = {
  badgeType: 'category',
  badge: dummyBadgeTemplate(1, 'ダミーカテゴリー')
};

const dummyTagBadge = {
  badgeType: 'tag',
  badge: dummyBadgeTemplate(1, 'ダミータグ')
};

const dummyPagination = {
  currentPage: 1,
  maxPage: Math.ceil(dummyArticles.length / 5)
};

const dummyTocs = dummyFactory(MAX_TOC_DATA, i => dummyTocTemplate(i));

const dummyLogo = 'ダミーロゴ';

export {
  dummyImg,
  dummyCategoryBadge,
  dummyTagBadge,
  dummyTagBadges,
  dummyDate,
  dummyArticles,
  dummyPagination,
  dummyClassificationCategory,
  dummyClassificationTag,
  dummyLogo,
  dummyTocs,
  dummyRoutes
};
