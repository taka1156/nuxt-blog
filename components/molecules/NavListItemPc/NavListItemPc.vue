<template>
  <div>
    <div class="nav-item">
      <base-link :route-to="navItem.to">
        <div class="nav-item__contents">
          <base-img
            :size="`sm`"
            :img-url="navItem.img"
            :img-alt="`${navItem.name}の画像`"
          />
          <base-text>{{ navItem.name }}</base-text>
        </div>
      </base-link>
    </div>
  </div>
</template>

<script>
import BaseImg from '../../atoms/BaseImg/BaseImg';
import BaseText from '../../atoms/BaseText/BaseText';
import BaseLink from '../../atoms/BaseLink/BaseLink';

export default {
  name: 'NavListItemPc',
  components: {
    'base-img': BaseImg,
    'base-text': BaseText,
    'base-link': BaseLink
  },
  props: {
    /**
     * ナビゲーションの１項目分
     */
    navItem: {
      type: Object,
      required: true
    }
  }
};
</script>

<style scoped>
::v-deep .base-text--extend {
  color: white;
}

.nav-item {
  display: block;
  font-size: 18px;
  text-align: left;
}

.nav-item__contents {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 100%;
  padding: 0;
  margin: 0;
}
</style>
