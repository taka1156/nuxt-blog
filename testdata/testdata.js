const dummyImg = {
  imgUrl: 'http://placehold.jp/150x150.png',
  imgAlt: 'ダミーの画像'
};

const dummyCategoryBadge = {
  badgeType: 'category',
  badge: {
    id: 1,
    name: 'ダミーカテゴリー',
    img: { url: 'http://placehold.jp/150x150.png' }
  }
};

const dummyTagBadge = {
  badgeType: 'tag',
  badge: {
    id: 1,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  }
};

const dummyTagBadges = [
  {
    id: 1,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  },
  {
    id: 2,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  },
  {
    id: 3,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  },
  {
    id: 4,
    name: 'ダミータグ',
    img: { url: 'http://placehold.jp/150x150.png' }
  }
];

const dummyDate = {
  createdAt: '2020-07-04T10:53:40.252Z',
  updatedAt: '2020-07-08T15:15:07.668Z'
};

const dummyMarkdown = `
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。** 
## この文章はダミーです。 
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
## この文章はダミーです。
  **文字の大きさ、量、字間、行間等を確認するために入れています。**
`;

const dummyArticles = [
  {
    id: '1pbb3xp8x',
    createdAt: '2020-09-13T14:28:35.489Z',
    updatedAt: '2020-09-13T16:01:29.403Z',
    publishedAt: '2020-09-13T14:28:35.489Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'hcpeu85l7',
    createdAt: '2020-09-13T12:38:43.014Z',
    updatedAt: '2020-09-13T15:55:21.517Z',
    publishedAt: '2020-09-13T12:38:43.014Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'hcpeu85l7',
    createdAt: '2020-09-13T12:38:43.014Z',
    updatedAt: '2020-09-13T15:55:21.517Z',
    publishedAt: '2020-09-13T12:38:43.014Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'hcpeu85l7',
    createdAt: '2020-09-13T12:38:43.014Z',
    updatedAt: '2020-09-13T15:55:21.517Z',
    publishedAt: '2020-09-13T12:38:43.014Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'hcpeu85l7',
    createdAt: '2020-09-13T12:38:43.014Z',
    updatedAt: '2020-09-13T15:55:21.517Z',
    publishedAt: '2020-09-13T12:38:43.014Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: '7zcsnuihh',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'xizlxccff',
    createdAt: '2020-09-07T04:35:28.817Z',
    updatedAt: '2020-09-07T12:59:56.511Z',
    publishedAt: '2020-09-07T04:35:28.817Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'piymdvgeb',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: 'aamcmuhtf',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      },
      {
        id: 'xghqpsmpp',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'pgxhtaaa9z',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  },
  {
    id: 'b-zlnpffc',
    createdAt: '2020-07-04T12:24:08.429Z',
    updatedAt: '2020-09-07T13:00:12.392Z',
    publishedAt: '2020-07-04T12:24:08.429Z',
    title:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています',
    summary:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量',
    body: dummyMarkdown,
    tags: [
      {
        id: 'svdq1mmtc',
        name: 'ダミー',
        img: {
          url: 'http://placehold.jp/150x150.png'
        }
      }
    ],
    category: {
      id: 'eq7p_6jei',
      name: 'ダミー',
      img: {
        url: 'http://placehold.jp/150x150.png'
      }
    }
  }
];

const dummyPagination = {
  currentPage: 1,
  maxPage: Math.ceil(dummyArticles.length / 5)
};

const dummyClassfication = [
  {
    id: 'xfrbr-upq',
    name: 'ダミー',
    img: {
      url: 'http://placehold.jp/150x150.png'
    }
  },
  {
    id: 'tlx_v85px',
    name: 'ダミー',
    img: {
      url: 'http://placehold.jp/150x150.png'
    }
  },
  {
    id: 'xghqpsmpp',
    name: 'ダミー',
    img: {
      url: 'http://placehold.jp/150x150.png'
    }
  },
  {
    id: 'dhyw5n7iw',
    name: 'ダミー',
    img: {
      url: 'http://placehold.jp/150x150.png'
    }
  }
];

const dummyLogo = 'ダミー';

const dummyRoutes = [
  {
    name: 'ダミー',
    to: '/test1',
    img: 'http://placehold.jp/150x150.png'
  },
  {
    name: 'ダミー',
    to: '/test2',
    img: 'http://placehold.jp/150x150.png'
  },
  {
    name: 'ダミー',
    to: '/test3',
    img: 'http://placehold.jp/150x150.png'
  },
  {
    name: 'ダミー',
    to: '/test4',
    img: 'http://placehold.jp/150x150.png'
  }
];

export {
  dummyImg,
  dummyCategoryBadge,
  dummyTagBadge,
  dummyTagBadges,
  dummyDate,
  dummyArticles,
  dummyPagination,
  dummyClassfication,
  dummyLogo,
  dummyRoutes
};
