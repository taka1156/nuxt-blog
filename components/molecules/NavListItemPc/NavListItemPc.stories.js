import NavListItemPc from './NavListItemPc.vue';
import { dummyRoutes } from '@/__testdata__/testdata.js';

export default {
  title: 'Molecules/NavListItemPc',
  component: NavListItemPc,
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  }
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { NavListItemPc },
  template: '<nav-list-item-pc v-bind="$props" />'
});

export const Default = Template.bind({});
Default.args = { navItem: dummyRoutes[0] };
