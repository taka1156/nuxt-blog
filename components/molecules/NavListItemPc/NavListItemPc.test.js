import { mount, RouterLinkStub } from '@vue/test-utils';
import NavListItemPc from './NavListItemPc.vue';
import { dummyRoutes } from '@/__testdata__/testdata.js';

describe('NavListItem', () => {
  const navListItemPc = mount(NavListItemPc, {
    stubs: {
      NuxtLink: RouterLinkStub
    },
    propsData: {
      navItem: dummyRoutes[0]
    }
  });

  it('NavList初期値: navItem', () => {
    // NavItme
    expect(navListItemPc.vm.$options.props.navItem.required).toBe(true);
    expect(navListItemPc.vm.navItem).toBe(dummyRoutes[0]);
  });

  it('値がDOMに反映されているか', () => {
    const aTag = navListItemPc.findComponent(RouterLinkStub);
    const imgTag = navListItemPc.find('img');

    expect(aTag.props().to).toBe(dummyRoutes[0].to);
    expect(aTag.text()).toBe(dummyRoutes[0].name);
    expect(imgTag.attributes().src).toBe(dummyRoutes[0].img);
    // スナップショット
    expect(navListItemPc.html()).toMatchSnapshot();
  });
});
