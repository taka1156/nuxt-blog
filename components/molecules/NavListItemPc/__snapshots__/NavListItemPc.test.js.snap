// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavListItem 値がDOMに反映されているか 1`] = `
"<div>
  <div class=\\"nav-item\\"><a class=\\"base-link--extend\\">
      <div class=\\"nav-item__contents\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img--extend base-img--sm\\">
        <p class=\\"base-text--extend\\">ダミーナビゲーション</p>
      </div>
    </a></div>
</div>"
`;
